### ISSUE TITLE
Dynamic component imports (lazy-loaded routes) don't work

### ISSUE THREAD

**Issue by iignatov** [vuejs/vue-router-next#757]
### Version
4.0.3

### Reproduction link
[https://codesandbox.io/s/dank-forest-sbzsh?file=/src/router.js](https://codesandbox.io/s/dank-forest-sbzsh?file=/src/router.js)

### Steps to reproduce
1. Add dynamic component import to a route, e.g.:
```
    {
      path: "/about",
      component: () => import("../views/About.vue")
    }
```
2. Click on the route link.
3. Observe the error in the console.

### What is expected?
The component should be loaded dynamically and displayed.

### What is actually happening?
```text
Unhandled error during execution of native event handler 
  at <RouterLink to="/about" > 
  at <App>
```
![vue-router-next-dynamic-import-error](https://user-images.githubusercontent.com/820242/107156155-4c80ac00-6985-11eb-86e0-bcef68e94161.png)

**Comment by posva** [comment]
You have a typo `component: () => import("./views/About.vue")`, you should see an error coming from webpack failing to resolve the path

**Comment by iignatov** [comment]
@posva Thank you for the quick reply!

This was actually only a mistake in the reproduction/demo environment. The import path in my actual repo was correct. However, while I was trying to update the demo environment, I found the culprit.

It turned out that I forgot to make the component class the default export, e.g. I had this:
```js
export class About extends Vue {
  render() {
    // TSX
  }
}
```

The issue was fixed after changing it to:
```js
export default class About extends Vue {
  render() {
    // TSX
  }
}
```

Hopefully this helps someone, who makes the same mistake.
